controllers/helloworld_controller.go

```diff
@@ -72,24 +63,72 @@ func (r *HelloworldReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	stopReconcile, result, err := r.validateOrCreateSvcFor(helloworld, ctx, req)
+
+	if stopReconcile {
+		return result, err
+	}
+
+	stopReconcile, result, err = r.validateOrCreatePodsFor(helloworld, ctx, req)
+
+	if stopReconcile {
+		return result, err
+	}
+
+	Logging(req, "Finished succesfully")
+
+	return ctrl.Result{Requeue: false}, nil
+}
+
+func (r *HelloworldReconciler) validateOrCreateSvcFor(helloworld *v1alpha1.Helloworld, ctx context.Context, req ctrl.Request) (bool, ctrl.Result, error) {
```

Comment:
I think this function should only return (serviceExists bool, err error)
and the calling function should then return "ctrl.Result" to match the result of this func.
What I mean in general is that this is an inner function, so it should be independent - and less aware of the fact that it's inside a controller and all.